src/Symfony/Component/Cache/Traits/FilesystemTrait.php

```diff
@@ -23,6 +23,29 @@
     use FilesystemCommonTrait;
 
     /**
+     * @return bool
+     */
+    public function prune()
+    {
+        $time = time();
+        $pruned = true;
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->directory, \FilesystemIterator::SKIP_DOTS), \RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if (!$h = @fopen($file, 'rb')) {
+                continue;
+            }
+
+            if ($time >= (int) $expiresAt = fgets($h)) {
+                $pruned = isset($expiresAt[0]) && @unlink($file) && !file_exists($file) && $pruned;
+            }
+
+            fclose($h);
```

Comment:
that won't work on Windows, where you cannot unlink a file when all handlers to it are not closed